src/plugins/github/api.js

```diff
@@ -0,0 +1,153 @@
+// @flow
+
+import {Graph} from "../../core/graph";
+import type {Node} from "../../core/graph";
+import type {Address} from "../../core/address";
+import type {
+  NodePayload,
+  EdgePayload,
+  NodeType,
+  IssueNodePayload,
+  PullRequestNodePayload,
+  CommentNodePayload,
+  AuthorNodePayload,
+  AuthorSubtype,
+} from "./types";
+
+import {
+  CONTAINS_EDGE_TYPE,
+  COMMENT_NODE_TYPE,
+  AUTHORS_EDGE_TYPE,
+  AUTHOR_NODE_TYPE,
+  ISSUE_NODE_TYPE,
+  PULL_REQUEST_NODE_TYPE,
+} from "./types";
+
+export class Repository {
+  repositoryName: string;
+  graph: Graph<NodePayload, EdgePayload>;
+
+  constructor(repositoryName: string, graph: Graph<NodePayload, EdgePayload>) {
+    this.repositoryName = repositoryName;
+    this.graph = graph;
+  }
+
+  issueOrPRByNumber(number: number): ?(Issue | PullRequest) {
+    let result: Issue | PullRequest;
+    this.graph.nodes({type: ISSUE_NODE_TYPE}).forEach((n) => {
+      if (n.payload.number === number) {
+        result = new Issue(this.graph, n.address);
+      }
+    });
+    this.graph.nodes({type: PULL_REQUEST_NODE_TYPE}).forEach((n) => {
+      if (n.payload.number === number) {
+        result = new PullRequest(this.graph, n.address);
+      }
+    });
+    return result;
+  }
+
+  issues(): Issue[] {
+    return this.graph
+      .nodes({type: ISSUE_NODE_TYPE})
+      .map((n) => new Issue(this.graph, n.address));
+  }
+
+  pullRequests(): PullRequest[] {
+    return this.graph
+      .nodes({type: PULL_REQUEST_NODE_TYPE})
+      .map((n) => new PullRequest(this.graph, n.address));
+  }
+
+  authors(): Author[] {
+    return this.graph
+      .nodes({type: AUTHOR_NODE_TYPE})
+      .map((n) => new Author(this.graph, n.address));
+  }
+}
+
+class GithubEntity<T: NodePayload> {
```

Comment:
It is somewhat strange that a repository is not a `GithubEntity`. I suppose that this is because it is not actually a node in the graph… should it be? (No change requested.)

Reply:
Having a repository node makes a lot of sense; we're almost sure to need them. Made an issue to track this suggestion. https://github.com/sourcecred/sourcecred/issues/171